### ISSUE TITLE
jenkins build still fails

### ISSUE THREAD

**Issue by marc-hanheide** [strands-project/strands_ci#8]
... and I don't understand the problem...

Here's what catkin_make says:

```
CMake Error: An attempt was made to access a variable: PCL_DIR that has not been defined. This variable is created by the FIND_PACKAGE command. CMake version 1.6 always converted the variable name to upper-case, but this behavior is no longer the case.  To fix this you might need to set the cache value of CMAKE_BACKWARDS_COMPATIBILITY to 1.6 or less.  If you are writing a CMake listfile, you should change the variable reference to use the case of the argument to FIND_PACKAGE.
```

It seems to have something to do with the new OpenNI because only since we include it it fails. I again suspect some missing dependencies there, but I don't really know. Does anybody have any ideas? @hawesie @kunzel @cdondrup @AlexanderHermans @Jailander or any other cmake guru?

**Comment by marc-hanheide** [comment]
must be some sort of parallelisation issue... when I run catkin_make a 2nd time it seems to work...

**Comment by cdondrup** [comment]
The only difference between this CMakeLists and the CMakeLists files in our navigation fork which also use pcl is that they use REQUIRED. This would at least cause a different error if cmake cannot find pcl at all. So just try `find_package(PCL REQUIRED)` and see if it changes anything.
This is the file I compared it with: https://github.com/strands-project/ros-navigation-fork/blob/groovy-devel-catkinized/base_local_planner/CMakeLists.txt

But there also is a difference in the package.xml. Try and use: `<build_depend>pcl_ros</build_depend>` and `<run_depend>pcl_ros</run_depend>` which might be the solution to the problem.

**Comment by marc-hanheide** [comment]
@cdondrup that didn't do the trick :-( See http://lcas.lincoln.ac.uk/jenkins/job/catkin_make_strands_desktop_full/113/console Still complains about the same thing and I checked it uses the right code. Any other idea?

**Comment by RaresAmbrus** [comment]
Hi,
I'm trying to view the Jenkins log but it says I'm lacking the reading permissions; can you please enable those?
Also, this is a strange error - the new openni package doesn't use PCL. But PCL does have its own openni wrapper which conflicts with the one that ROS provides.

**Comment by marc-hanheide** [comment]
@RaresAmbrus did you log in to jenkins? It should use your github account. Can you not log in at all at http://lcas.lincoln.ac.uk/jenkins or just not see that log?

**Comment by RaresAmbrus** [comment]
@marc-hanheide it seems I am logged in, but this is the error that I get: 

Access Denied

RaresAmbrus is missing the Overall/Read permission

**Comment by marc-hanheide** [comment]
OK, I figured it out, also by using @Jailander s account. You have to be a _public_ member of `strands-project` to be able to use jenkins. I.e. you must not conceal your membership (Why would you, you'd be proud of it wouldn't you ;-) ) Goto https://github.com/strands-project?tab=members and click on "publicize membership" behind your name.

**Comment by RaresAmbrus** [comment]
@marc-hanheide  Thanks for the update, I hadn't noticed that option until now ... And I'm able to view the jenkins build error now, I'll take a look and see if I can figure out what's causing it.

**Comment by marc-hanheide** [comment]
@RaresAmbrus have you made any progress on this? I want to get it up and running again, and indeed this is not only a problem in jenkins, but also shows up whenever one builds a system from scratch (at least for me).

**Comment by RaresAmbrus** [comment]
Hi @marc-hanheide 
I tried a couple of times last week but couldn't replicate it. Can you please tell me which repos you are checking out?All of them or just some specific ones?

**Comment by marc-hanheide** [comment]
@RaresAmbrus It's using desktop-full, as in doing `wstool init src https://raw.github.com/strands-project/strands_systems/master/strands_rosinstall/strands-desktop-full.yaml
` in your empty catkin workspace and then trying to build that with `catkin_make`

**Comment by RaresAmbrus** [comment]
Ok, I'm getting the same error. I'm looking into it right now, I guess one of the packages is calling 

```
find_package(PCL 1.6 REQUIRED)
```

without first defining PCL_DIR. Could that be the issue?

**Comment by marc-hanheide** [comment]
yes, I suppose so, but setting `PCL_DIR` to something hard-coded is clearly not correct. That's what cmake would be for, to find PCL. What's the point of `find_package` if people hard-code that path in their `CMakeLists.txt`? I wonder what the correct solution is...

**Comment by RaresAmbrus** [comment]
I'm not sure, but adding the `PCL_DIR` definition didn't remove the error. I'm removing packages one by one to see which is the one causing the cmake error.

**Comment by marc-hanheide** [comment]
This seems relevant: http://answers.ros.org/question/81306/catkin-cant-find-pclconfigcmake/

**Comment by RaresAmbrus** [comment]
This is strange - if I remove scitos_3d_mapping from the packages downloaded I don't get the PCL_DIR error anymore ... However, scitos_3d_mapping on its own compiles just fine. I'll take a look inside scitos_3d_mapping to see what's causing this.

**Comment by marc-hanheide** [comment]
As I said, I believe this a problem of parallel build. Because it always
works on a second build attempt.

On 5 December 2013 13:05, RaresAmbrus notifications@github.com wrote:

> This is strange - if I remove scitos_3d_mapping from the packages
> downloaded I don't get the PCL_DIR error anymore ... However,
> scitos_3d_mapping on its own compiles just fine. I'll take a look inside
> scitos_3d_mapping to see what's causing this.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/strands-project/strands_ci/issues/8#issuecomment-29895766
> .

## 

---

Dr. Marc Hanheide
Senior Lecturer in Computer Science
Room MC3221B
University of Lincoln
Brayford Pool
Lincoln, LN6 7TS
Phone: +44 1522 88 6966
EMail: marc@hanheide.net
Web: http://www.hanheide.net
skype: mah99de

**Comment by RaresAmbrus** [comment]
I think I found a fix. 
In `scitos_3d_mapping/scitos_ptu_sweep/CMakeLists.txt` replace this line:

```
find_package(catkin REQUIRED COMPONENTS actionlib actionlib_msgs message_generation rospy sensor_msgs pcl pcl_ros tf message_filters)
```

with this:

```
find_package(catkin REQUIRED COMPONENTS actionlib actionlib_msgs message_generation rospy sensor_msgs pcl_ros tf message_filters)
find_package(PCL 1.6 REQUIRED)
```

The pcl package needs to be included through a separate find_package call. It's a separate dependency and not a ros package. 

@marc-hanheide can you please try it out on your system and see if the issue is resolved ? It works for me now.

**Comment by marc-hanheide** [comment]
OK, that seems to work! Thanks, @RaresAmbrus ! Now we have another issue:
`In file included from /var/lib/jenkins/jobs/catkin_make_strands_desktop_full/workspace/catkin_ws/src/scitos_apps/scitos_docking/src/CPtuClient.cpp:1:0:
/var/lib/jenkins/jobs/catkin_make_strands_desktop_full/workspace/catkin_ws/src/scitos_apps/scitos_docking/src/CPtuClient.h:11:45: fatal error: scitos_apps_msgs/ChargingAction.h: No such file or directory
compilation terminated.
` 

Again, seems to be a problem with the parallel build and I don't quite see why this is happening either. Maybe @gestom can shed some light on this one? I believe the problem is that the dependencies in https://github.com/strands-project/scitos_apps/blob/master/scitos_docking/CMakeLists.txt are not correctly set for e.g. `add_library(CPtuClient src/CPtuClient.cpp)` It never says that this library (why is it a library after all) is dependent on any message generation, right? @gestom? I'll open a new issue in scitos-apps.

**Comment by gestom** [comment]
Hi, 
I have played  with the docking dependencies and now, the jenkins build fails on:

make[2]: **\* No rule to make target `/usr/lib/x86_64-linux-gnu/libQt5OpenGL.so.5.0.2', needed by`/var/lib/jenkins/jobs/catkin_make_strands_desktop_full/workspace/catkin_ws/devel/lib/scitos_mira/scitos_node'. Stop.

**Comment by marc-hanheide** [comment]
Ok, that is a problem related to the fact that qt5 is also installed on that machine, I'll investigate. It's annoying how qt deals with different versions being installed in parallel. I'll see what I can do.

**Comment by marc-hanheide** [comment]
yeah! back online!
